scipy/spatial/tests/test_distance.py

```diff
@@ -398,28 +398,37 @@ def setup_method(self):
                               'int': [np.float32, np.double],
                               'float32': [np.double]}
 
-    def test_cdist_extra_args(self):
+    def test_cdist_extra_args(self, metric):
         # Tests that args and kwargs are correctly handled
+
+        X1 = [[1., 2., 3.], [1.2, 2.3, 3.4], [2.2, 2.3, 4.4]]
+        X2 = [[7., 5., 8.], [7.5, 5.8, 8.4], [5.5, 5.8, 4.4]]
+        kwargs = {'N0tV4l1D_p4raM': 3.14, "w": np.arange(3)}
+        args = [3.14] * 200
+
+        assert_raises(TypeError, cdist, X1, X2,
```

Comment:
Since we are at it, we should match all of these with the message that would be raised.

`with pytest.raises(TypeError, match=r"Message...")`

Reply:
I started looking at the error messages here, and the first three distances I tried gave me three different messages (conceptually). Adding a `match=` clause can & should be done in a later PR.

Reply:
github seems to have eaten on of your comments about this (did I respond? 🤔), but in any case, the messages were too divergent to try to unify them even with regular expressions - leaving this for a follow-up.